### PR TITLE
Now I can play with Aya in VSCode happily

### LINKED ISSUE #291: Goto def. for module and import commands
🤔

### PR DESCRIPTION
**Pull request by imkiva**
This PR improved existing features:
- Binding emission logic for better hover messages
- Find references in pattern matchings and projections
- Position resolver now correctly finds all `Var`s (we cannot resolve ctors, fields and local telescopes if the cursor was placed at the name part of their definitions before)

and introduced new features:
- Rename a symbol
- Highlight all occurrences of a text by scoping rule (instead of string comparison!!)
- Goto definition for import commands (close #291)

and generalized some tools for future IDE development:
- The `Resolver` class which handles concrete term interactions (all features mentioned above heavily depend on this _busybox_)
- The `ModuleVar` for expanding the definition-based mechanism to modules (like goto module, rename module, etc.)

### CODE REVIEW

**Comment by ice1000** [comment]
> Position resolver now correctly finds all Vars (we cannot resolve ctors, fields and local telescopes if the cursor was placed at the name part of their definitions before)

I didn't quite get this part, can you give me an example snippet where the old code didn't work?

File: lsp/src/main/java/org/aya/lsp/actions/GotoDefinition.java [hunk 1]
```diff
@@ -19,18 +26,42 @@
...
-      var res = LspRange.toLoc(from, target);
-      if (res != null) Log.d("Resolved: %s in %s", target, res.getTargetUri());
-      return res;
-    }).collect(Collectors.toList());
+      return new WithPos<>(from, target);
+    });
+  }
+
+  private static @Nullable SourcePos mockSourcePos(@NotNull SeqView<LibraryOwner> libraries, @NotNull ModuleVar moduleVar) {
+    var src = Resolver.resolveModule(libraries, moduleVar.path().ids()).getOrNull();
```
Comment (ice1000) [inline lsp/src/main/java/org/aya/lsp/actions/GotoDefinition.java:62]: You can use `.map.getOrNull`

**Review by ice1000 — APPROVED** [review]

**Comment by imkiva** [comment]
> > Position resolver now correctly finds all Vars (we cannot resolve ctors, fields and local telescopes if the cursor was placed at the name part of their definitions before)
> 
> I didn't quite get this part, can you give me an example snippet where the old code didn't work?

Imagine the code
```aya
open data Nat
  | zero
  | suc Nat
```
Ctrl-click the `suc`, we provide nothing before.

**Comment by ice1000** [comment]
@imkiva IC, thanks!

**Comment by ice1000** [comment]
Congratulations!! You've created the 300-th ticket in this repository

**Pull request merged**